### DISCUSSION TITLE
Using the query builder independently from the client

### DISCUSSION THREAD

**Discussion by bencompton** [prisma/prisma#8912] [Q&A]
I'm curious whether there is any possibility of using the query builder separately from the client to build strongly typed queries and then output the SQL text for a particular DB engine. All of the docs and examples I've seen show the client and query builder as the same API, but I'm wondering if they can be used independently somehow, even if it involves consuming internals of this package.

For context, my goal would be to use the query builder in the browser with SQL.js for development, and also use SQL.js for integration tests, while using MySQL from Node.js everywhere else (see #5385). I understand that the Prisma client currently depends on Rust code and doesn't currently run in the browser, so SQL.js is a non-starter for the Prisma client. However, not sure if the query builder itself is entirely TypeScript that could run in the browser and generate SQL text, which could then be fed to either SQL.js, or when running in Node.js, the Prisma client's `executeRaw`, or even another DB client library (e.g., [mysql](https://www.npmjs.com/package/mysql)).

**Comment by janpio** [comment]
No, unfortuantely the whole query builder is part of the Rust engine so there is no way to access that. Also, many of the generated queries depend on the results of the first query - so without running them, you can not have the full set of queries to run.

**Reply by bencompton** [reply]
Thanks for the insights! That explains why I didn't get far combing through the TypeScript.

I generally prefer more control over SQL anyhow, but mainly would use a query builder for SQL injection mitigation and the ability to generate SQL for multiple DB engines. I really like Prisma's capabilities around generating type definitions for schemas and managing migrations, so I might end up building my own minimalistic query builder for basic CRUD operations that leverages Prisma's generated types.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!

**Comment by aleclarson** [comment]
Now that Prisma is migrating away from the Rust engine, is this feature a possibility again?
